src/stackset.ts

```diff
@@ -655,7 +663,17 @@ export class StackSet extends Resource implements IStackSet {
       permissionModel: deploymentTypeConfig.permissionsModel,
       callAs: deploymentTypeConfig.callAs,
       templateUrl: props.template.templateUrl,
-      stackInstancesGroup: Lazy.any({ produce: () => { return this.stackInstances; } }),
+      stackInstancesGroup: Lazy.any({
+        produce: () => {
+          return this.stackInstances;
+        },
+      }),
+      parameters: props.parameters ? Object.entries(props.parameters).map((entry) => {
+        return {
+          parameterKey: entry[0],
+          parameterValue: entry[1],
+        };
+      }) : undefined,
```

Comment:
this will add `parameters: undefined` to everyone's stackset, which isn't the end of the world, but people don't like snapshot diffs. instead we can do the following pattern:

```ts
...(props.parameters ? { parameters: {} } : {})
```

this way no parameters means no parameter property at all and no one's diff gets needlessly updated.